### PR TITLE
set default for kpk in storage delegation

### PR DESCRIPTION
**Pull request by Aayush-Abhyarthi**
### Description

Issue: https://github.com/terraform-ibm-modules/terraform-ibm-watsonx-saas-da/issues/153

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
@Aayush-Abhyarthi The upgrade test does not have KMS encryption enabled, so we won't be able to test this change in the upgrade test. This is a gap we probably wan't to plug before proceeding with this PR

File: tests/pr_test.go [hunk 1]
```diff
@@ -178,3 +178,87 @@ func TestWithExistingKP(t *testing.T) {
...
+					"module.configure_user.null_resource.configure_user",
+					"module.configure_user.null_resource.restrict_access",
+				},
+			},
+			TerraformVars: map[string]interface{}{
+				"location":            validRegions[rand.Intn(len(validRegions))],
+				"resource_group_name": prefix,
+				"provider_visibility": "public",
+				"cos_kms_crn":         terraform.Output(t, existingTerraformOptions, "key_protect_crn"),
+				"cos_kms_key_crn":     terraform.Output(t, existingTerraformOptions, "kms_key_crn"),
```
Comment (ocofaigh) [inline tests/pr_test.go:243]: To enable encryption, you need to set `enable_cos_kms_encryption = true`. The fact that the test passed even when passing values for `cos_kms_crn` and `cos_kms_key_crn` makes me think we should add variable validation to the DA code to ensure that if user does pass values for these, that `enable_cos_kms_encryption` has to be set to true

Comment (ocofaigh) [reply tests/pr_test.go:243]: It also looks like `TestWithExistingKP` also needs this update

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: tests/pr_test.go (hunk 1 above)
Comment (Aayush-Abhyarthi) [reply tests/pr_test.go]: @ocofaigh Added `enable_cos_kms_encryption = true` , still the test passed.

File: tests/pr_test.go [hunk 2]
```diff
@@ -178,3 +179,88 @@ func TestWithExistingKP(t *testing.T) {
...
+					"module.configure_user.null_resource.configure_user",
+					"module.configure_user.null_resource.restrict_access",
+				},
+			},
+			TerraformVars: map[string]interface{}{
+				"location":                  validRegions[rand.Intn(len(validRegions))],
+				"resource_group_name":       prefix,
+				"provider_visibility":       "public",
+				"enable_cos_kms_encryption": true,
+				"cos_kms_crn":               terraform.Output(t, existingTerraformOptions, "key_protect_crn"),
```
Comment (ocofaigh) [inline tests/pr_test.go:244]: @Aayush-Abhyarthi This is why upgrade test is not failing - your passing the DA an existing key, so its not creating a new key, so it never uses the `cos_kms_new_key_name` input

**Comment by ocofaigh** [comment]
@Aayush-Abhyarthi we can't proceed with this change - its a breaking change:
```
        	Messages:   	Resource(s) identified to be destroyed 
        	            	Name: kms_key
        	            	Address: module.storage_delegation[0].ibm_kms_key.kms_key[0]
        	            	Actions: [create delete]
```

This is why I wanted to make sure we had a test to catch this. I think we cannot proceed with this change. Since we are breaking down this DA into individual DAs, its probably ok to just leave it as is

File: tests/pr_test.go [hunk 3]
```diff
@@ -178,3 +179,87 @@ func TestWithExistingKP(t *testing.T) {
...
+			},
+		})
+
+		output, err := options.RunTestUpgrade()
+		assert.Nil(t, err, "This should not have errored")
+		assert.NotNil(t, output, "Expected some output")
```
Comment (ocofaigh) [inline tests/pr_test.go:250]: This needs to be updated to:

```
        output, err := options.RunTestUpgrade()
	if !options.UpgradeTestSkipped {
		assert.Nil(t, err, "This should not have errored")
		assert.NotNil(t, output, "Expected some output")
	}
```

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comment on why upgrade test is not working

**Comment by ocofaigh** [comment]
when updating, please also unskip the upgrade test (use a commit message with the text `UNSKIP UPGRADE TEST`)

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**